File: frontend/ufinet_frontend/src/types/Car.ts
export interface CarRequest{
    brand: string;
    model: string;
    color: string;
    year: number
    carPlate:string;
    imageUrl: string;
}

export interface CarResponse{
    id: number;
    brand: string;
    model: string;
    color: string;
    year: number;
    carPlate:string;
    imageUrl: string;

}

File: backend/technical_test/src/main/java/com/example/technical_test/entity/Car.java
package com.example.technical_test.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "cars")
public class Car {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(name = "brand", nullable = false )
    private String brand;
    @Column(name = "model", nullable = false )
    private String model;
    @Column(name = "year", nullable = false)
    private int year;
    @Column(name = "color", nullable = false)
    private String color;
    @Column(name = "car_plate", nullable = false, unique = true)
    private String carPlate;
    @Column(name = "image_url")
    private String imageUrl;
    @ManyToOne(fetch =  FetchType.LAZY)
    @JoinColumn(name="user_id", nullable = false)
    private User user;
}
